src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeColorization.cs

```diff
@@ -0,0 +1,199 @@
+            WindowBackgroundManager.UpdateBackground(currentWindow, ApplicationTheme.Dark, WindowBackdropType.Mica, false);
+        }
+        else if(IsThemeHighContrast() && Utility.IsOSWindows11OrNewer) 
+        {
+            if(themeToApply.Contains("hcwhite"))
+            {
+                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hcwhite.xaml", UriKind.Absolute));
+                WindowBackgroundManager.UpdateBackground(currentWindow, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
+            }
+            else if(themeToApply.Contains("hcblack"))
+            {
+                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hcblack.xaml", UriKind.Absolute));
+                WindowBackgroundManager.UpdateBackground(currentWindow, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
+            }
+            else if (themeToApply.Contains("hc1"))
+            {
+                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hc1.xaml", UriKind.Absolute));
+                WindowBackgroundManager.UpdateBackground(currentWindow, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
+            }
+            else
+            {
+                UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "hc2.xaml", UriKind.Absolute));
+                WindowBackgroundManager.UpdateBackground(currentWindow, ApplicationTheme.HighContrast, WindowBackdropType.None, false);
+            }
+        }
+        else if (Utility.IsOSWindows11OrNewer)
+        {
+            UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "light.xaml", UriKind.Absolute));
+            WindowBackgroundManager.UpdateBackground(currentWindow, ApplicationTheme.Light, WindowBackdropType.Mica, false);
+        }
+
+        _currentApplicationTheme = themeToApply;
+        _currentAppsTheme = GetAppsTheme();
+    }
+
+    /// <summary>
+    /// Fetches registry value
+    /// </summary>
+    /// <returns>string indicating the current theme</returns>
+    internal static string GetSystemTheme()
+    {
+        string systemTheme = Registry.GetValue(
+            _regThemeKey,
+            "CurrentTheme",
+            "aero.theme"
+            ) as string
+            ?? String.Empty;
+
+        return systemTheme;
+    }
+
+    internal static string GetAppsTheme()
+    {
+        return IsThemeDark() ? "Dark" : "Light";
+    }
+    #endregion
+
+
+    internal static bool IsThemeDark()
```

Comment:
As noted in the comments before, this should check the background color values as per recommendation. If this is not available, `AppsUseLightTheme` is fine, but it also should fallback to `SystemUsesLightTheme` and possibly other fallbacks.

Reply:
We had already changed the logic to `AppsUseLightTheme`'s value. I have now, also addressed the fallback to `SystemUsesLightTheme`.